Design app fixes after Webpacker installation

ferblape opened the pull request:
#### :tophat: What? Why?

This PR fixes a couple of issues related with #7733 

1. The app needed to be updated to use Webpacker new configuration, thus the files have been added
2. The README and some docs need to be updated accordingly
3. A test has been added to check some of the webpacker configuration (the package.json and it's lock file) are in synch with Decidim

As a summary, now the process is a bit long:

1. bundle install
2. npm install
3. bin/rails s
4. (optional) bin/webpack-dev-server

#### :pushpin: Related Issues

- Related to #7733

#### Testing

You should be able to see the design app working again 😅 

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](https://github.com/decidim/decidim/blob/develop/CONTRIBUTING.adoc) to this repository.

- [ ] :question: **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] :heavy_check_mark: **DO** check open PR's to avoid duplicates.
- [ ] :heavy_check_mark: **DO** keep pull requests small so they can be easily reviewed.
- [ ] :heavy_check_mark: **DO** build locally before pushing.
- [ ] :heavy_check_mark: **DO** make sure tests pass.
- [ ] :heavy_check_mark: **DO** make sure any new changes are documented in `docs/`.
- [ ] :heavy_check_mark: **DO** add and modify seeds if necessary.
- [ ] :heavy_check_mark: **DO** add CHANGELOG upgrade notes if required.
- [ ] :heavy_check_mark: **DO** add to GraphQL API if there are new public fields.
- [ ] :heavy_check_mark: **DO** add link to MetaDecidim if it's a new feature.
- [ ] :x:**AVOID** breaking the continuous integration build.
- [ ] :x:**AVOID** making significant changes to the overall architecture.

### :camera: Screenshots
*Please add screenshots of the changes you're proposing*
![Description](URL)

:hearts: Thank you!

mrcasals commented:
Actually, @ferblape, why not just commit the files? Wouldn't that be safer for anyone using this app? It feels strange that I need to manually run the command to install Webpacker to the application when the app is already there 😕

ferblape commented:
I see your point, it's similar to the Gemfile case, both the design app and decidim share the Gemfile and it's commited, right?

I didn't want to commit it to avoid the files to be outdated, but I guess that's something that can be prevented with a test or a check in the release process. What do you think?

mrcasals commented:
YEah, sounds good!

ferblape commented:
@mrcasals ready to review again. PR description updated with the new approach and updated docs.

Also, please check a test I've added in system/ to verify package.json is synchronized 😄

mrcasals commented:
Nice, looks good to me!
